openwisp_controller/connection/tests/test_models.py

```diff
@@ -1008,29 +1007,33 @@ def _assert_applying_conf_test_command(mocked_exec):
             # exit code 1 considers the update not successful
             self.assertEqual(conf.status, 'modified')
 
-    @mock.patch.object(update_config, 'delay')
-    def test_device_update_config_in_progress(self, mocked_update_config):
+    @mock.patch('time.sleep')
+    def test_device_update_config_in_progress(self, mocked_sleep):
         conf = self._prepare_conf_object()
 
         with mock.patch('celery.app.control.Inspect.active') as mocked_active:
             mocked_active.return_value = {
                 'task': [{'name': _TASK_NAME, 'args': [str(conf.device.pk)]}]
             }
+            conf.config = {'general': {'timezone': 'UTC'}}
+            conf.full_clean()
             conf.save()
             mocked_active.assert_called_once()
-            mocked_update_config.assert_not_called()
+            mocked_sleep.assert_called_once()
 
-    @mock.patch.object(update_config, 'delay')
-    def test_device_update_config_not_in_progress(self, mocked_update_config):
+    @mock.patch('time.sleep')
+    def test_device_update_config_not_in_progress(self, mocked_sleep):
         conf = self._prepare_conf_object()
 
         with mock.patch('celery.app.control.Inspect.active') as mocked_active:
             mocked_active.return_value = {
                 'task': [{'name': _TASK_NAME, 'args': ['...']}]
             }
+            conf.config = {'general': {'timezone': 'UTC'}}
+            conf.full_clean()
             conf.save()
             mocked_active.assert_called_once()
-            mocked_update_config.assert_called_once_with(conf.device.pk)
+            mocked_sleep.assert_called_once()
```

Comment:
What exactly are we testing with this assertion?

Reply:
https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/connection/tasks.py#L27

It's not a critical change.

Before the `update_config` celery task was mocked because the logic called by this test was executed in the receiver function, now everything is executed in the celery task.

Reply:
`test_device_update_config_not_in_progress` and `test_device_update_config_in_progress` have become identical after this change. We should also check the `DeviceConnection.update_config() `is not executed when the upgrade is in progress.